### DISCUSSION TITLE
Using the backslash character to do decimal division

### DISCUSSION THREAD

**Discussion by Sasino97** [dotnet/csharplang#4420] [Language Ideas]
I suggest making the \ character usable to perform decimal division between int, without having to cast them.

Example:
```csharp
var x = 17 / 2;
var y = 17 \ 2;
```
x is an `int` and it contains 8
y is a `decimal` and it contains 8.5m

**Comment by svick** [comment]
* Why? `17 \ 2` is only a single character shorter then `17m / 2`, but it's much less clear.
* As much as I like `decimal`, I think `double` is used much more often. So why should there be special operator just for `decimal`?
* There aren't many characters that can be easily written on almost any keyboard. I don't think C# should waste one of them on something like this.

**Reply by bernd5** [reply]
I think with _decimal devision_ he does not mean the type `decimal` in particular - he means floating-point devision in general.

As far as I understand, he does not want `int / int -> int` - he wants e.g. `int \ int -> double | decimal?`.
But you are right - to do such a division we can simply use floating-point literals (like `17m` or `17d`).

I think it is not worth to introduce such an operator.
